crmd/utils.c

```diff
@@ -1161,3 +1161,36 @@ void crmd_peer_down(crm_node_t *peer, bool full)
     crm_update_peer_join(__FUNCTION__, peer, crm_join_none);
     crm_update_peer_expected(__FUNCTION__, peer, CRMD_JOINSTATE_DOWN);
 }
+
+gboolean
+crmd_is_rsc_managed(const char *rsc)
+{
+    int offset = 0;
+    static int xpath_max = 1024;
+    gboolean rv = TRUE;
+    xmlNode *xml_search = NULL;
+    char *xpath_string = NULL;
+    const char *managed = NULL;
+
+    xpath_string = calloc(1, xpath_max);
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "%s", get_object_path("resources"));
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "//*[@id=\"%s\"]", rsc);
+    offset += snprintf(xpath_string + offset, xpath_max - offset, "//nvpair[");
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "@name=\"%s\"", XML_RSC_ATTR_MANAGED);
+    offset += snprintf(xpath_string + offset, xpath_max - offset, "]");
+
+    if (fsa_cib_conn->cmds->query(fsa_cib_conn, xpath_string, &xml_search,
+            cib_sync_call | cib_scope_local | cib_xpath) == pcmk_ok) {
+        managed = crm_element_value(xml_search, XML_NVPAIR_ATTR_VALUE);
+        if (managed && (crm_is_true(managed) == FALSE)) {
+            rv = FALSE;
+        }
+    }
+
+    free(xpath_string);
+    free_xml(xml_search);
+    return rv;
+}
```

Comment:
I do think we should have a complete story for what unmanaged means for a remote connection, whether that's caused by the administrator or by conditions. It's not simple to distinguish how or why a resource is unmanaged.

I'm not sure how to get the information to the crmd. Any information from the PE would have to be via the transition graph.